src/lib.rs

```diff
@@ -162,18 +165,26 @@ impl AstarteSdk {
             )
             .await?;
 
-        for i in ifaces {
-            self.client
-                .subscribe(
-                    self.client_id() + "/" + interface::traits::Interface::name(&i.1) + "/#",
-                    rumqttc::QoS::ExactlyOnce,
-                )
-                .await?;
+        for (_, iface) in ifaces_filter {
+            self.subscribe_server_owned_interface(&iface).await?;
         }
 
         Ok(())
     }
 
+    async fn subscribe_server_owned_interface(
+        &self,
+        iface: &Interface,
+    ) -> Result<(), AstarteError> {
```

Comment:
As it is this assumes the caller correctly passes just server owned interfaces, it would probably be a good idea to check it also here and either return an error if the interface is not server owned or print a warning and avoid subscribing to it